python/mxnet/autograd.py

```diff
@@ -112,6 +137,34 @@ def pause(is_train=False):
     return RecordingStateScope(False, is_train)
 
 
+def override_train():
```

Comment:
Yes, it's equivalent, but the point is on how to use it. To use action-based API, user needs to keep track of the training/recording state by themselves, whereas a scope-based API does it for you. This is the difference between a sequence of get and save current state => set state => revert back the state afterwards, versus a simple scope.

Also, imagine that in the model codes you want to do different things based on whether it's training or predicting. You can't do that with action-based API without involving a book keeping for recording status.